src/daemon/jsonrpc/api.rs

```diff
@@ -103,41 +101,60 @@ impl RpcApi for RpcImpl {
     fn listvaults(
         &self,
         meta: Self::Metadata,
-        status: Option<String>,
+        statuses: Option<Vec<String>>,
         txids: Option<Vec<String>>,
     ) -> jsonrpc_core::Result<serde_json::Value> {
-        let status = if let Some(status) = status {
-            Some(VaultStatus::from_str(&status).map_err(|_| {
-                JsonRpcError::invalid_params(format!("'{}' is not a valid vault status", &status))
-            })?)
+        let statuses = if let Some(statuses) = statuses {
+            // If they give an empty array, it's not that they don't want any result, but rather
+            // that they don't want this filter to be taken into account!
+            if statuses.len() > 0 {
+                Some(
+                    statuses
+                        .into_iter()
+                        .map(|status_str| {
+                            VaultStatus::from_str(&status_str).map_err(|_| {
+                                JsonRpcError::invalid_params(format!(
+                                    "'{}' is not a valid vault status",
+                                    &status_str
+                                ))
+                            })
+                        })
+                        .collect::<jsonrpc_core::Result<Vec<VaultStatus>>>()?,
+                )
+            } else {
+                None
+            }
         } else {
             None
         };
         let txids = if let Some(txids) = txids {
```

Comment:
Ithink that if you unwrap an option to return an option, use map
```
 let txids = if let Some(txids) = txids {...} else {None};
```
to 
```
let txids = txids.map(|txids|{...});
```

Reply:
You can't return an error in a closure.. 
```rust
.collect::<jsonrpc_core::Result<Vec<VaultStatus>>>()?,
```
https://github.com/re-vault/revaultd/pull/46/files/881fa57932918a3046c1afb71d438c0f53cbad55#diff-5416085ff6b3b242b02325c6140edced6cff30147ef1b8d1fc2c30b6b0a23c2bR122